src/lexer.coffee

```diff
@@ -790,7 +790,7 @@ exports.Lexer = class Lexer
     LINE_CONTINUER.test(@chunk) or
     @tag() in ['\\', '.', '?.', '?::', 'UNARY', 'MATH', 'UNARY_MATH', '+', '-',
                '**', 'SHIFT', 'RELATION', 'COMPARE', '&', '^', '|', '&&', '||',
-               'BIN?', 'THROW', 'EXTENDS']
+               'BIN?', 'THROW', 'EXTENDS', 'DEFAULT']
```

Comment:
Out of interest, but would adding 'RETURN" here achieve the same thing as https://github.com/jashkenas/coffeescript/pull/4532/commits/75f6912529f3f8f546dbdb9e5210bfa22fd61cd0?